### PR TITLE
Added functionality for networking field in config file

### PR DESCRIPTION
**Pull request by JonathanNolan3**
Allows for the addition of a networking field in the config file which contains IP Addresses. These addresses can be used to add more listening addresses and external broadcasting addresses.
Example config file attached
[serverConfig.json](https://github.com/user-attachments/files/16956330/serverConfig.json)

### CODE REVIEW

**Comment by JonathanNolan3** [comment]
Still needs changes on how to implement using field internalBroadcast

File: instrumentserver/apps.py [hunk 1]
```diff
@@ -77,15 +76,17 @@ def serverScript() -> None:
...
     else:
         serverWithGui(port=args.port,
                       addresses=args.listen_at,
                       initScript=args.init_script,
                       serverConfig=serverConfig,
                       stationConfig=stationConfig,
                       guiConfig=guiConfig,
-                      pollingThread=pollingThread)
+                      pollingThread=pollingThread,
+                      ipAddresses=ipAddresses)
```
Comment (marcosfrenkel) [inline instrumentserver/apps.py:89]: Keep singular term `externalAddress` or something like that throughout the entire class

**Comment by marcosfrenkel** [comment]
Just as an FYI, that formatting makes it a yaml file with yml extension

File: instrumentserver/apps.py (hunk 1 above)
Comment (marcosfrenkel) [reply instrumentserver/apps.py:89]: Talked in person

**Comment by marcosfrenkel** [comment]
3 Things before merging:

- [ ] Add documentation on what ipAdresses are, what the field it expects and how it should be used.
- [ ] Have the period of broadcast accept floats.
- [ ] add a logging call in the beginning that says if it has started broadcasting on a broadcast address (a message saying, no external broadcast if this is not the case)

**Comment by JonathanNolan3** [comment]
Added another commit, should have fixed all three of the above

**Review by marcosfrenkel — APPROVED** [review]

**Comment by marcosfrenkel** [comment]
Merged.

@JonathanNolan3 could you please do #94 as soon as possible?

**Pull request merged**